### DISCUSSION TITLE
Allow for discount codes / allocations to be present on the cart object

### DISCUSSION THREAD

**Discussion by ricksnp** [Shopify/function-examples#359] [Feedback: Cart and Checkout Validation API]
Currently, we have an employee discount powered by Shopify Scripts. 
- If user has a specific tag, they get a percentage discount

However, these users should not be able to use discount codes. I'd like to see if they entered discount codes, and prevent them from checking out. Otherwise, I could possibly use a CheckoutUI extension to remove a discount code based on the user's email domain. 

To add, I could also do a no code solution by making sure all of our Marketing/Loyalty apps create discounts for specific customer segments, to where I don't even need to have this check in place. 

Also open to recommendations or if I just re-read the API!!

**Comment by meandillar** [comment]
Bump! We really need this also.

**Comment by maks-afew** [comment]
Bump!

**Comment by itsdikey** [comment]
Bump!

**Comment by Kaniwong** [comment]
It’s been almost a year pass and this feature is really important. We need to disable discount when has bundle in cart like we did it in Shopify script

**Comment by juliana-zhu** [comment]
we are trying to switch from shopify script to shopify function, but currently we are unable to obtain the actual discount applied in the current shopping cart and the amount that the end user should actually pay. We hope to do some discounts with full gifts and full discounts. We hope to be able to Only by getting the discount information currently applied by the user in the shopping cart can we judge whether the full discount can be applied. I am very frustrated. Now I don’t know how to achieve it.

**Reply by juliana-zhu** [reply]
It seems that the shopify function is an action executed before the checkout is created, while the shopify script is an action executed after the checkout is created, so it is difficult to obtain some attributes after the checkout is generated.
